### ISSUE TITLE
just works with WiFi

### ISSUE THREAD

**Issue by salimkayabasi** [mauimauer/cheapcast#8]
I tested on some devices
nexus 7 
nexus 4
galaxy nexus
UG007

works with all device
only works with android to android casting
no connection established with chrome to android

UG007 which has ethernet port
when I connect this device on ethernet, ChromeCast cannot connect.
I have to change it to Wifi

**Comment by kaazvaag** [comment]
I have the same issue.

I want to use ethernet instead of WiFi.

**Comment by jayasafunctionofe** [comment]
I'm seeing the following unhandled exception in logcat when starting CheapCast on an ethernet connection:

```
D/PreferenceActivity( 1032): Connected to CheapCastService
W/NotificationTrackerService(  315): Dropping notification from at.maui.cheapcast
E/WifiStateMachine(  315): Error! unhandled message{ what=131157 when=0 }
W/System.err( 1058): 2013-08-13 23:45:55.675:INFO:oejs.Server:jetty-8.y.z-SNAPSHOT
W/System.err( 1058): 2013-08-13 23:45:55.688:INFO:oejs.AbstractConnector:Started SelectChannelConnector@0.0.0.0:8008
D/CheapCastService( 1058): Initialized HTTP/WS Server
D/CheapCastService( 1058): Initialized SSDP/DIAL Discovery
```

Does this show up for everyone else having issues on ethernet as well?

**Comment by kaazvaag** [comment]
No, also when I connect through ethernet to the device 192.168.1.108:8008 I get the message: This is CheapCast :D.

So the cheapcast app is running just fine (and is reachable through ethernet), but it is not able to give a "cheapcast" signal through ethernet (because it only shows the chromecast symbol on youtube/music when putting the Minix X5 in the WiFi only mode).

Any idea how to fix this?

**Comment by jayasafunctionofe** [comment]
@kaazvaag, if you go to http://192.168.1.108:8008/ssdp/device-desc.xml look for the section "URLBase", does it show "http://127.0.0.1:8008" ? 

If so, this is why it's not showing up as an available device. I believe the app is failing to properly acquire the IP address when using an ethernet connection,

**Comment by kaazvaag** [comment]
@ja450n You are right. 

When I connect with ethernet (fixed IP 192.168.1.108) and enable CheapCast I get an URLBASE http://127.0.0.1:8008.
When I connect with WiFi only (fixed IP 192.168.1.109) mode and enable CheapCast I get an URLBASE http://192.168.1.109:8008.

Next step. How to fix this issue?

**Comment by jayasafunctionofe** [comment]
@kaazvaag based on the exception occurring in the WifiStateMachine, i'm going to have to guess that this is a bug only the developer can fix.

**Comment by gregmac** [comment]
Just confirmed same issue with a Pivos Xios DS
